src/middlewared/middlewared/service.py

```diff
@@ -907,10 +908,19 @@ def ping_remote(self, options):
         Method that will send an ICMP echo request to "hostname"
         and will wait up to "timeout" for a reply.
         """
-        if options['type'] == 'ICMP' or options['type'] == 'ICMPV4':
-            ping_host = self._ping_host(options['hostname'], options['timeout'])
-        elif options['type'] == 'ICMPV6':
-            ping_host = self._ping6_host(options['hostname'], options['timeout'])
```

Comment:
So If I specify ICMPV6 your code may resolve to ipv4 and will just erroneously work